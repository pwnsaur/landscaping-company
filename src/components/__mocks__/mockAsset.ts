import { Asset } from 'contentful';

export const createMockAsset = (): Asset => {
  return {
    sys: {
      id: '1',
      type: 'Asset',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'asset',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'jonathan-kemper-kYl3Yp2hwtQ-unsplash',
      description: '',
      file: {
        url: '//images.ctfassets.net/sandie-clarke-q13Zq1Jufks-unsplash.jpg',
        details: {
          size: 500939,
          image: {
            width: 1920,
            height: 1080,
          },
        },
        fileName: '',
        contentType: 'image/jpeg',
      },
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Asset),
  };
};
