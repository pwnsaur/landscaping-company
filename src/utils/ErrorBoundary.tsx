import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
