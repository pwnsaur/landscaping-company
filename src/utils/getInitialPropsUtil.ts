import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
