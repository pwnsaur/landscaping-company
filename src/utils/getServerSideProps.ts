import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
