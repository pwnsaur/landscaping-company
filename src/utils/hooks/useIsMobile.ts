import useWindowSize from './useWindowSize';

const useIsMobile = (): boolean => {
  const { width, height } = useWindowSize();
  const currentWidth = width || 0;
  const currentHeight = height || 0;
  const isMobile = currentWidth <= 640;

  return isMobile;
};

export default useIsMobile;
