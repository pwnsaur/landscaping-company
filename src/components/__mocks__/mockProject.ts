import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeProjectFields } from '@/types/contentfulTypes';

export const createMockProject = (): Entry<TypeProjectFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'project',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Fifteenth Project',
      slug: 'fifteenth-project',
      content: {
        nodeType: BLOCKS.DOCUMENT,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
      date: '2023-04-27',
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeProjectFields>),
    update: () => Promise.resolve({} as Entry<TypeProjectFields>),
  };
};
