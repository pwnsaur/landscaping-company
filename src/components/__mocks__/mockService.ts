import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeServiceFields } from '@/types/contentfulTypes';

export const createMockService = (): Entry<TypeServiceFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'service',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Service',
      slug: 'service',
      excerpt: 'service',
      description: {
        nodeType: BLOCKS.PARAGRAPH,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeServiceFields>),
    update: () => Promise.resolve({} as Entry<TypeServiceFields>),
  };
};
