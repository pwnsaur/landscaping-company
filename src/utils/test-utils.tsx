import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
