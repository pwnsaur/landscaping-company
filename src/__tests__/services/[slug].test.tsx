import 'jest-styled-components';

import { createMockService } from '@/components/__mocks__/mockService';
import Service, {
  getStaticPaths,
  getStaticProps,
} from '@/pages/services/[slug]';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render } from '@/utils/test-utils';

jest.mock('../../utils/hooks/useIsMobile');

const mockService = createMockService();

describe('Service page', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<Service service={mockService} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have correct styling for desktop', () => {
    (useIsMobile as jest.Mock).mockReturnValue(false);

    const { container } = render(<Service service={mockService} />);
    const projectContainer = container.firstChild;
    expect(projectContainer).toHaveStyleRule('width', '70%');
  });

  test('should have correct styling for mobile', () => {
    (useIsMobile as jest.Mock).mockReturnValue(true);

    const { container } = render(<Service service={mockService} />);
    const projectContainer = container.firstChild;
    expect(projectContainer).toHaveStyleRule('width', '100%');
  });

  test('should export getStaticPaths function', () => {
    expect(getStaticPaths).toBeDefined();
  });

  test('should export getStaticProps function', () => {
    expect(getStaticProps).toBeDefined();
  });
});
