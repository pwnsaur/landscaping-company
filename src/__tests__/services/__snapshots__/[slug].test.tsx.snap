// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service page should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 70%;
  margin: 3rem 5rem;
}

.c2 {
  font-size: 1.5rem;
  font-weight: 600;
  margin-bottom: 1rem;
}

.c1 {
  width: 100%;
  height: auto;
  object-fit: cover;
  margin: 1rem 0;
}

.c3 {
  margin: 1rem 0;
  width: 80%;
}

<div
    class="c0"
  >
    <img
      alt="cover image"
      class="c1"
      data-nimg="1"
      decoding="async"
      height="1080"
      src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=3840&q=75 2x"
      style="color: transparent;"
      width="1920"
    />
    <h2
      class="c2"
    >
      Service
    </h2>
    <div
      class="c3"
    >
      <p />
    </div>
  </div>
</DocumentFragment>
`;
